app/assets/javascripts/omni_nav/google-custom-search.js

```diff
@@ -0,0 +1,314 @@
+      var hasSearchFilters = $container.find('.search-filter-option').length;
+      if ( hasSearchFilters ) $selectedSearchFilter.text(DEFAULT_FILTER_TEXT);
+    };
+
+    var renderGoogleSearchMarkup = function() {
+      // Each GCS element must have a unique id to use as its gname.
+      var searchBoxDiv = $searchBox[0];
+      var searchResultsDiv = $searchResults[0];
+
+      var searchBoxConfig = {
+        gname: containerId,
+        div: searchBoxDiv,
+        tag: 'searchbox',
+        attributes: {
+          enableAutoComplete: true,
+          autoCompleteMatchType: 'any',
+          resultSetSize: 6,
+          enableHistory: false
+        }
+      };
+
+      var searchResultsConfig = {
+        gname: containerId,
+        div: searchResultsDiv,
+        tag: 'searchresults',
+        attributes: {
+          linkTarget: '_self',
+          enableOrderBy: true
+        }
+      };
+
+      google.search.cse.element.render(searchBoxConfig, searchResultsConfig);
+    };
+
+    var updateSearchResultsHeight = function() {
+      $searchResultsContainer.height($(window).height());
+    };
+
+    var onSearchEnter = function(e) {
+      // The autocomplete container is not present until the user starts typing
+      // Add click event listener the first time it shows up
+      bindAutocomplete();
+
+      var resultsRequested = e.type == 'click' || e.which == ENTER_KEY;
+      var searchTermsPresent = gcsElement.getInputQuery().length > 0;
+      if( resultsRequested && searchTermsPresent ) {
+        showResults();
+      }
+    };
+
+    var onSearchFilterClick = function() {
+      $selectedSearchFilter.text($(this).text());
+    };
+
+    var onSearchResultsResize = function() {
+      clearTimeout(resizeTimeoutId);
+      resizeTimeoutId = setTimeout(updateSearchResultsHeight, 500);
+    };
+
+    var bindAutocomplete = function() {
+      // Autocomplete map. This seems very fragile as it requires Google to use
+      // the same hardcorded selector each time. But it is how this was originally coded
+      // and so far has proved reliable so I am keeping it for now.
+      var contextSelectorMap = {
+        // omniNavContext: selector
+        'primary': 'table.gstl_50.gssb_c',
+        'utility': 'table.gstl_51.gssb_c'
+      };
+
+      // Bind only once: use a data flag to track binding.
+      var autoCompleteIsBound = $container.data('autoCompleteIsBound');
+      if ( autoCompleteIsBound ) return;
+
+      // This is required for selecting the Autocomplete block associated with
+      // SearchComponent. Since these are added post-loading by Google and are attached
+      // as absolute elements directly to the body, the parent cannot be deduced
+      // from the object itself. Checked Google API. No accessor. This is unfortunate.
+      var omniNavContext = ( containerId.indexOf("utility") >= 0 ) ? 'utility' : 'primary';
+      var autoCompleteSelector = contextSelectorMap[omniNavContext];
```

Comment:
I really don't like this. But I don't think there's a clear better option at this point. Especially as we still consider eliminating the two forms and have the `OmniNav2` module switch one back and forth depending on view dimensions.